forth.asm

```diff
@@ -203,6 +203,28 @@ dict_entry tonumber, '>NUMBER', 0x0 ;( str -- n)
     push    ebx
     jmp     next
 
+interpret_if:
+    ; if stack nozero, advance instruction pointer
+    pop     eax
+    test    eax, eax
+    jnz     .true
+    add     esi, 0x8
+    jmp     next
+.true:
+    jmp     next
+
+
+dict_entry interpret, 'INTERPRET-WORD', 0x0 ; ( str -- )
+    call    docolon
+    dd      blank
+    dd      cword
+    dd      find
+    dd      interpret_if
```

Comment:
Consider making that inline assembly code into a proper word.  Perhaps it would consume a stack element, and if the stack element is non-zero, it adds some number to the Forth PC.